### ISSUE TITLE
not all outputs in a glob expression recognized

### ISSUE THREAD

**Issue by dvoet** [broadinstitute/cromwell#1633]
I have a few cases in Firecloud (using cromwell .19) where it appears like some scatter jobs are not being launched. I have tracked it down to a problem with outputs from the prepare call not being registered as actual outputs. According to the logs and the gcs bucket all the outputs from the prepare step are right, but according to cromwell the output array is missing a couple entries.  Looking in [workflow metadata](https://github.com/broadinstitute/cromwell/files/557271/metadata.txt) at the CallingGroup_Workflow.mone_prepare call, the outputs for split_base.06.interval_list and split_base.22.interval_list are missing. According to the [google bucket listing of the glob dir](https://github.com/broadinstitute/cromwell/files/557270/glob_listing.txt), the files are there. The JES logs for the prepare step show the upload happens successfully.

To summarize, the call CallingGroup_Workflow.mone_prepare successfully generates and upload 25 interval files but the output section of the metadata shows only 23 (6 and 22 are missing). This means that only 23 downstream calls run which is bad.

[workflow log](https://github.com/broadinstitute/cromwell/files/557272/workflow.9b9e6b19-aac4-451f-bf24-a9132c0e5408.txt)

**Comment by geoffjentry** [comment]
If this is what I think it is (an eventual consistency issue in google), this has been resolved in 21+

**Comment by dvoet** [comment]
I was thinking eventual consistency too because a bucket listing will not always be right.

**Comment by dvoet** [comment]
out of curiosity, how is it solved in 21?

**Comment by geoffjentry** [comment]
I take that back, it's not yet fixed although the fix is possible: #1395

**Comment by cjllanwarne** [comment]
Closing as duplicate. See also #1640
